crates/core/src/logstore/storage/mod.rs

```diff
@@ -98,22 +98,88 @@ impl ObjectStoreRegistry for DefaultObjectStoreRegistry {
     }
 }
 
-#[derive(Debug, Clone, Default)]
+#[derive(Debug, Clone, Default, DeltaConfig)]
 pub struct LimitConfig {
+    #[delta(alias = "concurrency_limit", env = "OBJECT_STORE_CONCURRENCY_LIMIT")]
     pub max_concurrency: Option<usize>,
 }
 
-impl config::TryUpdateKey for LimitConfig {
-    fn try_update_key(&mut self, key: &str, v: &str) -> DeltaResult<Option<()>> {
-        match key {
-            // The number of concurrent connections the underlying object store can create
-            // Reference [LimitStore](https://docs.rs/object_store/latest/object_store/limit/struct.LimitStore.html)
-            // for more information
-            "OBJECT_STORE_CONCURRENCY_LIMIT" | "concurrency_limit" => {
-                self.max_concurrency = Some(config::parse_usize(v)?);
+#[cfg(test)]
+#[allow(drop_bounds, unused)]
+mod tests {
+    use std::collections::HashMap;
+    use std::env;
+
+    use rstest::*;
```

Comment:
Seems like its been too long that I looked into test frameworks for rust. Learned about `rstest` with this PR and I think using fixtures and other mechanics it offers, we can greatly reduce some of our boilerplate.

The kernel PR also adds dat tests for rust, which I want to carve out and use as an opportunity to maybe refactor some of our test scaffolding.